txclib/parsers.py

```diff
@@ -43,253 +81,349 @@ def delete_parser():
+    main_parser.add_argument("-l", "--language", action="store",
+                             dest="language", default=None,
+                             help="Specify the source language of the "
+                             "resource")
+    main_parser.add_argument("-t", "--type", action="store", dest="i18n_type",
+                             help=("Specify the i18n type of the resource(s). "
+                                   "This is only needed, if the resource(s) "
+                                   "does not exist yet in Transifex. For a "
+                                   "list of available i18n types, see "
+                                   "http://docs.transifex.com/formats/"
+                                   ))
+    return main_parser
+
+
+def set_extra_parser():
+    extra_parser = ArgumentParser(add_help=False)
+    extra_parser.add_argument("--minimum-perc", action="store",
+                              dest="minimum_perc",
+                              help=("Specify the minimum acceptable "
+                                    "percentage of a translation in "
+                                    "order to download it."))
+    extra_parser.add_argument(
         "--mode", action="store", dest="mode", help=(
             "Specify the mode of the translation file to pull (e.g. "
             "'reviewed'). See http://bit.ly/pullmode for the "
             "available values."
         )
     )
-    group = OptionGroup(parser, "Extended options", "These options can only "
-                                "be used with the --auto-local command.")
-    group.add_option("-s", "--source-language", action="store",
-                     dest="source_language", default=None,
-                     help="Specify the source language of a resource "
-                     "[requires --auto-local].")
-    group.add_option("-f", "--source-file", action="store", dest="source_file",
-                     default=None, help="Specify the source file of a "
-                     "resource [requires --auto-local].")
-    group.add_option("--execute", action="store_true", dest="execute",
-                     default=False, help="Execute commands "
-                     "[requires --auto-local].")
-    parser.add_option_group(group)
+    return extra_parser
+
+
+def set_parser(subparser=False):
+    """Return the command-line parser for the set command."""
+    description = "This command can be used to create a mapping between "\
+        "files and projects either\nusing local files or using files from a "\
+        "remote Transifex server."
+    epilog = "\nSubcommands:\n"\
+        "    auto-local\n"\
+        "    auto-remote\n"\
+        "    bulk\n\n"\
+        "Examples:\n"\
+        "To set the source file:\n\
+        $ tx set -r project.resource --source -l en <file>\n\n"\
+        "To set a single translation file:\n\
+        $ tx set -r project.resource -l de <file>\n"
+    auto_local_epilog = "\nExamples:\n"\
```

Comment:
Don't we need descriptions for `auto_local` and `auto_remote`? Epilogs are nice, but I think a description should be added.

Reply:
Ok, I will add a brief description for these as well.